erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -1507,12 +1498,42 @@ def set_status(self, update=False, status=None, update_modified=True):
 					self.status = "Paid"
 				else:
 					self.status = "Submitted"
+
+				if (
+					self.status in ("Unpaid", "Partly Paid", "Overdue")
+					and self.is_discounted
+					and get_discounting_status(self.name) == "Disbursed"
+				):
+					self.status += " and Discounted"
```

Comment:
There's no "Partly Paid and Discounted" option added in the sales invoice status.